lib/rules/disallow-accessing-ember-properties.js

```diff
@@ -0,0 +1,33 @@
+var assert = require('assert');
+
+function DisallowAccessingEmberProperties() { }
+
+DisallowAccessingEmberProperties.prototype = {
+  configure: function(value) {
+    assert(
+      value === true,
+      this.getOptionName() + ' option requires a true value or should be removed'
+    );
+  },
+
+  getOptionName: function() {
+    return 'disallowAccessingEmberProperties';
+  },
+
+  check: function(file, errors) {
+    file.iterateNodesByType('MemberExpression', function(node) {
+      if (node.parentNode.type === 'AssignmentExpression') {
+        // Avoid throwing the error if we're assigning to an Ember property
+        return;
+      }
+      if (node.object.name === 'Ember') {
+        var propertyName = node.property.name;
+        errors.add(
```

Comment:
I don't technically recall ever having to access things like `Ember.MODEL_FACTORY_INJECTIONS` to be honest, only setting them.

The problem with not raising an error when the property name is all caps is that the rule would ignore `Ember.RSVP`, `Ember.A`, `Ember.K`, etc. when those should really be destructured.